admin_light_user/models/res_users.py

```diff
@@ -0,0 +1,142 @@
+    """
+    seperators = tree.xpath("//separator")
+    group_administration = tree.xpath("//group[@string='Administration']")
+    for node in seperators + group_administration:
+        sibling_node = node.getnext()
+        if sibling_node is None or sibling_node.tag != "field":
+            node.getparent().remove(node)
+
+
+def _remove_admin_application_selection_fields(
+    tree: etree._Element, env: api.Environment
+):
+    """Remove selection fields of groups related to super admin applications from the view tree.
+
+    :param tree: the xml tree of the view to modify.
+    :param env: the odoo environment
+    """
+    admin_applications = [env.ref(app) for app in SUPER_ADMIN_APPLICATIONS]
+
+    selection_group_fields = tree.xpath("//field[starts-with(@name, 'sel_groups')]")
+    for node in selection_group_fields:
+        selected_group_ids = [
+            int(g) for g in node.attrib["name"].replace("sel_groups_", "").split("_")
+        ]
+        selected_groups = env["res.groups"].browse(selected_group_ids)
+        if selected_groups[0].category_id in admin_applications:
+            node.getparent().remove(node)
+
+
+class ResUsers(models.Model):
+
+    _inherit = "res.users"
+
+    @api.model
+    def create(self, vals):
+        if self.env.user.has_group("admin_light_user.group_user_management"):
+            self._check_not_granting_admin_priviledges(vals)
+
+        return super().create(vals)
+
+    def write(self, vals):
+        if self.env.user.has_group("admin_light_user.group_user_management"):
+            self._check_not_granting_admin_priviledges(vals)
+            self._check_not_unarchiving_admin(vals)
+
+        return super().write(vals)
+
+    def _check_not_granting_admin_priviledges(self, vals):
+        groups_id = vals.get("groups_id", [])
+
+        admin_group_ids = {self.env.ref(group_ref).id for group_ref in ADMIN_GROUPS}
+
+        is_granting_admin = any(
+            (command[0] == 4 and command[1] in admin_group_ids)
+            or (command[0] == 6 and set(command[2]) & admin_group_ids)
+            for command in groups_id
+        )
+
+        if is_granting_admin:
+            raise AccessError(
+                _(
+                    "You are not authorized to grant admin priviledges. "
+                    "Only the super admin can."
+                )
+            )
+
+    def _check_not_unarchiving_admin(self, vals):
+        if "active" in vals:
+            super_admin_group = self.env.ref("base.group_erp_manager")
+            if super_admin_group in self.mapped("groups_id"):
+                raise AccessError(
+                    _(
+                        "You are not authorized to archive or unarchive a user with "
+                        "super admin priviledges. "
+                        "Only the super admin can."
+                    )
+                )
+
+    @api.model
+    def update_admin_light_user_group_view(self):
```

Comment:
Check here if the user has not the group erp_manager and has group admin_light_user before proceding to changes.